--- src/Makefile.am~	2005-10-13 13:41:14.000000000 +0200
+++ src/Makefile.am	2009-02-14 05:04:14.000000000 +0100
@@ -33,7 +33,8 @@
 	-avoid-version \
 	-rpath $(moduledir) \
 	-module \
-	@LIBTOOL_EXPORT_OPTIONS@ \
+	@LIBTOOL_EXPORT_OPTIONS@
+prime_la_LIBADD = \
 	@SCIM_LIBS@
 
 if SCIM_BUILD_SETUP
@@ -54,7 +55,7 @@
 prime_imengine_setup_la_LDFLAGS = -avoid-version \
                                    -rpath $(setupmoduledir) \
                                    -module \
-                                   @LIBTOOL_EXPORT_OPTIONS@ \
-                                   @SCIM_GTKUTILS_LIBS@ \
+                                   @LIBTOOL_EXPORT_OPTIONS@
+prime_imengine_setup_la_LIBADD =   @SCIM_GTKUTILS_LIBS@ \
                                    @INTLLIBS@ \
                                    @SCIM_LIBS@
